### ISSUE TITLE
Release 2.12 validations checklist and cherry-picks

### ISSUE THREAD

**Issue by atalman** [pytorch/pytorch#182554]
# Release 2.12 validations checklist and cherry-picks

This issue tracks validation items and cherry-picks for the PyTorch 2.12 release.

Content is sourced from:
- [Milestone 2.12.0](https://github.com/pytorch/pytorch/milestone/63) open and closed issues (excluding [RFC] issues)
- [Release Tracker #180506](https://github.com/pytorch/pytorch/issues/180506) cherry-picks to the [release/2.12](https://github.com/pytorch/pytorch/tree/release/2.12) branch

Release branch CI status: [HUD](https://hud.pytorch.org/hud/pytorch/pytorch/release%2F2.12)

## High Priority

- [x] validate torchaudio 2.11 build cuda 13.2. Validation workflows: https://github.com/pytorch/test-infra/actions/runs/25443082712/job/74676558285 @NicolasHug 
- [x] [vllm] vllm::sparse_attn_indexer custom op output aliases input — now hard-errors with new torch.compile aliasing check (DeepSeek-V3.2) https://github.com/pytorch/pytorch/issues/182006 - @huydhn
- [x] [torch 2.12] umbrella issue - vLLM CI failures https://github.com/pytorch/pytorch/issues/180899 - @atalman
- [x] [hi-pri] 2026.04.24 Pytorch nightly wheel for Windows is non functional (LE: cuDNN DLLs are missing, manually adding them solves the issue) https://github.com/pytorch/pytorch/issues/181375 - @jovan2009
- [x] [vllm] [triton 3.7] PassManager::run failed in make_ttgir when compiling vLLM _penalties_kernel https://github.com/pytorch/pytorch/issues/180908 - @atalman
- [x] [hi-pri] [COMPILE] torch compile is broken with autograd.Function with completely incorrect outputs most of the times in PyTorch 2.11 https://github.com/pytorch/pytorch/issues/180642 - @mayank31398
- [x] [hi-pri] [ROCm][Release Blocker] All ROCm 7.1/7.2 nightly builds broken since March 29 https://github.com/pytorch/pytorch/issues/179524 - @atalman
- [x] [hi-pri] PyTorch CPU wheel index does not expose upload date metadata https://github.com/pytorch/pytorch/issues/178980 - @winwinashwin
- [ ] [hi-pri] MPS: `mm`/`addmm` SEGFAULTS on M4 if 2nd matrix is padded with `LORADOWN GEMV` https://github.com/pytorch/pytorch/issues/178056 - @malfet
- [x] [hi-pri] `torch.compile` fails to capture user-defined kernels with latest Triton https://github.com/pytorch/pytorch/issues/170049 | https://github.com/pytorch/pytorch/pull/172354 - @jjvraw

## MPS

- [ ] [MPS] scaled_dot_product_attention returns wrong values for non-contiguous (permute-produced) q/k/v https://github.com/pytorch/pytorch/issues/181133 | https://github.com/pytorch/pytorch/pull/181151 - @Isalia20
- [ ] MPS conv2d gives incorrect results for channel-slice views of channels_last tensors https://github.com/pytorch/pytorch/issues/180984 - @Tripton
- [ ] MPS autograd returns zero/corrupted batched input gradients for `torch.autograd.grad(y.sum(), x)` while `grad_outputs=torch.ones_like(y)` is correct https://github.com/pytorch/pytorch/issues/180201 - @txdai
- [ ] [MPS] Intermittent indexing out of bounds `AcceleratorError` https://github.com/pytorch/pytorch/issues/178079 - @hvaara
- [ ] [MPS] Fix bool mask handling in 1-pass SDPA decode kernel https://github.com/pytorch/pytorch/pull/182285 - @Isalia20
- [ ] [MPS] Fix nested ops.masked variable name collisions in Metal codegen https://github.com/pytorch/pytorch/pull/178304 - @malfet
- [ ] [cherry-pick] [MPS] Fix sliced channels_last tensors handling https://github.com/pytorch/pytorch/pull/181107 - @malfet
- [ ] [cherry-pick] [MPS] Fix SDPA wrong output for permuted q/k/v with B > 1 https://github.com/pytorch/pytorch/pull/181886 - @Isalia20
- [ ] [cherry-pick] [MPS] Fix bool mask handling in 1-pass SDPA decode kernel (#182285) https://github.com/pytorch/pytorch/pull/182311 - @malfet

## ROCm

- [x] revert https://github.com/pytorch/pytorch/pull/172340 https://github.com/pytorch/pytorch/pull/179151 - @nikitaved
- [x] [cherry-pick] [ROCm] Fix inline_asm_elementwise for ROCm https://github.com/pytorch/pytorch/pull/180600 - @chinmaydk99
- [x] [cherry-pick] [UT][ROCm][inductor] ROCm-specific XFAILS list for torchinductor_opinfo_property https://github.com/pytorch/pytorch/pull/180687 - @naromero77amd
- [x] [cherry-pick] [ROCm] Update scaled_mm DeepSeek error message https://github.com/pytorch/pytorch/pull/180690 - @jagadish-amd
- [x] [cherry-pick] [ROCm] Enable ROCm swizzle check and update scaled_mm swizzle tests https://github.com/pytorch/pytorch/pull/180691 - @jagadish-amd
- [x] [cherry-pick] [ROCm] Resolve timeouts caused due to hipblasLT module creation during graph capture https://github.com/pytorch/pytorch/pull/180692 - @chinmaydk99
- [x] [cherry-pick] [ROCm] Fix evaluate_platform_supports_fp8 false-positive https://github.com/pytorch/pytorch/pull/180715 - @jagadish-amd
- [x] [cherry-pick] [ROCm] Run test_scaled_mm_deepseek_error_messages on mi350 arch https://github.com/pytorch/pytorch/pull/180897 - @jagadish-amd
- [x] [cherry-pick] [ROCm][UT] Remove previously retained Triton 3.7 skip for torchinductor_opinfo test https://github.com/pytorch/pytorch/pull/180903 - @naromero77amd

## XPU

- [x] [cherry-pick] [xpu][fix] Include lazy_triton_compile.h in XPU cpp_wrapper header https://github.com/pytorch/pytorch/pull/180815 - @etaf
- [x] [cherry-pick] [xpu][fix] Enable bmm_outer_product Triton override for XPU https://github.com/pytorch/pytorch/pull/180816 - @etaf
- [x] [cherry-pick] [XPU][dynamo] Save XPU autocast state in dynamo global state https://github.com/pytorch/pytorch/pull/181349 - @weishi-deng

## Milestone Issues

- [x] [dynamo] fullgraph=True error on eager fallback has bad interaction with "eager_on_recompile" stance https://github.com/pytorch/pytorch/issues/181817 - @williamwen42
- [x] [Triton][Triton-Ext] Include TLX Triton Extension https://github.com/pytorch/pytorch/issues/178917 - @CRobeck
- [x] Some `test_sdpa_can_compile_dynamic` in `transformers` fail (with `torch.compile(dynamic=True)`) with `torch 2.12 RC` but pass with `torch 2.11` https://github.com/pytorch/pytorch/issues/181563 - @ydshieh
- [x] https://github.com/pytorch/pytorch/pull/179082 is BC-breaking https://github.com/pytorch/pytorch/issues/181122 - @zou3519
- [x] [CI][B200] Smoke Test Regressions https://github.com/pytorch/pytorch/issues/178787 - @nWEIdia
- [x] import torch fails with release 2.11 on python 3.13.8, AST parsing error (IndentationError) https://github.com/pytorch/pytorch/issues/178255 | https://github.com/pytorch/pytorch/pull/179066 - @malfet
- [x] `ncclDevCommCreate` in torch failed with NCCL 2.29.7 https://github.com/pytorch/pytorch/issues/177379 - @kwen2501
- [x] opaque objects don't work with constrain_to_fx_strides https://github.com/pytorch/pytorch/issues/175973 - @zou3519
- [x] [dynamo][guards] Disable recursive dict tag optimization https://github.com/pytorch/pytorch/pull/181873 - @anijain2305
- [x] [dynamo] warn instead of error on fullgraph=True fallback to eager for 2.12 https://github.com/pytorch/pytorch/pull/181818 - @williamwen42
- [x] Fix link to C++ torch stable docs https://github.com/pytorch/pytorch/pull/181604 - @janeyx99
- [x] Add TLS stack_bounds so we dont pay expensive reads https://github.com/pytorch/pytorch/pull/181018 - @drisspg
- [ ] Fix cholesky(upper=True) on macOS for matrices larger than block size https://github.com/pytorch/pytorch/pull/179154 - @malfet
- [x] Add cublas drop-in fix for 13.0  https://github.com/pytorch/pytorch/pull/178333 - @tinglvv
- [ ] [dynamo][einops] Suppress allow_in_graph deprecation warning https://github.com/pytorch/pytorch/pull/178324 - @anijain2305
- [x] [inductor] Defer input size/stride assertions to first use https://github.com/pytorch/pytorch/pull/177783 - @eellison
- [ ] [Inductor] Improve materialization heuristic for a chain of computations https://github.com/pytorch/pytorch/pull/176994 - @BoyuanFeng
- [ ] [inductor] symm_mem planning for graph inputs and fallback regions https://github.com/pytorch/pytorch/pull/175449 - @tianrengao
- [ ] [inductor] Lower functional symm_mem ops to ExternKernelOut for output buffer reuse https://github.com/pytorch/pytorch/pull/174856 - @tianrengao

## Cherry-picks

- [x] [cherry-pick] [dynamo] warn instead of error on fullgraph=True fallback to eager https://github.com/pytorch/pytorch/pull/181940 - @williamwen42
- [x] [cherry-pick] [dynamo][guards] Disable recursive dict tag optimization https://github.com/pytorch/pytorch/pull/181925 - @anijain2305
- [x] [cherry-pick] Enable fetch-tags in checkout-pytorch to fix release tag detection https://github.com/pytorch/pytorch/pull/180508 - @atalman
- [x] [cherry-pick] Conditionally skipped NCCL suspend/get_memory_stats/resume tests for NCCL lower than 2.29.7 https://github.com/pytorch/pytorch/pull/180693 - @albmalamd
- [x] [cherry-pick] [NCCL][Symmetric Memory] Fix groupName in IntraNodeComm https://github.com/pytorch/pytorch/pull/180809 - @Aidyn-A
- [x] [cherry-pick] Increase Python docs build timeout to 45m https://github.com/pytorch/pytorch/pull/180847 - @jeanschmidt
- [x] [cherry-pick] [user-streams] Fix cudagraphs compatibility with current stream (#180… https://github.com/pytorch/pytorch/pull/180916 - @mlazos
- [ ] [cherry-pick] [caching] Fix SymBool pickling issue with torch.cond. https://github.com/pytorch/pytorch/pull/180934 - @zhxchen17
- [ ] [cherry-pick] [dynamo] Filter aliased intermediates in autograd.Function forward tracing https://github.com/pytorch/pytorch/pull/180964 - @anijain2305
- [x] [cherry-pick] [dynamo] Fix cuda_stream pointer extraction for generic torch.Stream https://github.com/pytorch/pytorch/pull/181019 - @GSumanth109
- [x] [cherry-pick] Skip llms-full.txt during Sphinx build, generate in nightly push https://github.com/pytorch/pytorch/pull/181070 - @svekars
- [x] [cherry-pick] Add TLS stack_bounds so we dont pay expensive reads https://github.com/pytorch/pytorch/pull/181137 - @drisspg
- [x] [cherry-pick] Disable llms-full.txt https://github.com/pytorch/pytorch/pull/181141 - @svekars
- [ ] [cherry-pick] [Bugfix] Fix the MetaProxy error caused by skipping dispatch https://github.com/pytorch/pytorch/pull/181170 - @Lucaskabela
- [ ] [cherry-pick] Split onehot checks for CPU and accelerators https://github.com/pytorch/pytorch/pull/181211 - @Silv3S
- [ ] [cherry-pick] [inductor] makes cuda 13.0 cross compliation works (#179229) https://github.com/pytorch/pytorch/pull/181287 - @Gasoonjia
- [ ] [cherry-pick] [Bugfix] Fix triton crash in mixed device case for bmm https://github.com/pytorch/pytorch/pull/181294 - @Lucaskabela
- [ ] [cherry-pick] Disable dispatch modes when pickling AOTAutograd cache entries https://github.com/pytorch/pytorch/pull/181381 - @angelayi
- [ ] [cherry-pick] [Inductor] Improve materialization heuristic for a chain of computaions (#176994) https://github.com/pytorch/pytorch/pull/181441 - @azahed98
- [ ] [cherry-pick] [GraphPickler] Preserve FakeScriptObject for value-type opaques durin… https://github.com/pytorch/pytorch/pull/181454 - @angelayi
- [ ] [cherry-pick] [Inductor] Call latest c_shim version for versioned fallback ops https://github.com/pytorch/pytorch/pull/181548 - @manuelcandales
- [ ] [cherry-pick] [AOTI] Add BC-safe c_shim v2 for _scaled_dot_product_attention_math_for_mps enable_gqa https://github.com/pytorch/pytorch/pull/181549 - @manuelcandales
- [ ] [cherry-pick] Add enable_gqa parameter to SDPA MPS meta registration https://github.com/pytorch/pytorch/pull/181550 - @manuelcandales
- [x] [cherry-pick] [CD] Remove cu132->cu130 wheel install fallback in Dockerfile  https://github.com/pytorch/pytorch/pull/181577 - @atalman
- [ ] [cherry-pick] Cherry-pick #181161 and its forward fix #181434 https://github.com/pytorch/pytorch/pull/181608 - @janeyx99
- [x] [cherry-pick] Fix link to C++ torch stable docs https://github.com/pytorch/pytorch/pull/181613 - @janeyx99
- [x] [cherry-pick] Use full clone for docs build to fix nightly hang https://github.com/pytorch/pytorch/pull/181661 - @atalman
- [ ] [cherry-pick] [FSDP1] revert PR 178223 to bring back AG prefetching https://github.com/pytorch/pytorch/pull/181669 - @weifengpy
- [x] [cherry-pick] Add checkout-mode input to setup-linux action https://github.com/pytorch/pytorch/pull/181702 - @huydhn
- [x] [cherry-pick] [CI][Build] Goodbye Bazel https://github.com/pytorch/pytorch/pull/181721 - @malfet
- [x] [cherry-pick] [Bugfix][Caching] Fix Caching error by avoiding failure for triton kernels in pickling error (#181463) https://github.com/pytorch/pytorch/pull/181751 - @Lucaskabela
- [x] [cherry-pick] Fix dynamic shape tile issue (#181793) https://github.com/pytorch/pytorch/pull/181795 - @eellison
- [x] [cherry-pick] Make docs build behave the same for push=true and push=false https://github.com/pytorch/pytorch/pull/181921 - @svekars
- [x] [cherry-pick] Reduce sidebar navigation size for generated API pages https://github.com/pytorch/pytorch/pull/181943 - @svekars
- [x] [cherry-pick] Avoid raw stream name collisions in Inductor https://github.com/pytorch/pytorch/pull/182178 - @eellison
- [x] [cherry-pick] Fix stale PYTORCH_RELEASES_CODE_CC dict (fixes #182250) https://github.com/pytorch/pytorch/pull/182369 - @atalman
